Feature: reload endpoint for hot load changes from config file

antoniomerlin opened the pull request:
In this PR an endpoint is added for loading hot changes without restarting the container for #346 . Just making POST request to `/-/reload` latest config changes are available in metrics.

Let me know if any changes are required in this.

thomaspeitz commented on cmd/yace/main.go:
```diff
@@ -170,3 +171,22 @@ func (s *scraper) scrape(ctx context.Context) (err error) {
 	return nil
 }
+
+func (s *scraper) reload(ctx context.Context) func(http.ResponseWriter, *http.Request){
```
Lets put this function into the pkg/config.go. Would like to keep main.go small.

Can you please verify if the code was run in gofmt? https://pkg.go.dev/cmd/gofmt

antoniomerlin replied:
Hi thanks for reviewing, yes the code is run in gofmt

thomaspeitz reviewed (changes requested):
Thanks for your PR :) lets make some small changes and get this merged.

thomaspeitz commented:
@antoniomerlin If you get this fixed till Thuersday morning, I get this into the next release.

Else I schedule it for next week :)

Greetings :)

antoniomerlin commented:
Hi i removed that reload as separate function and added directly inside http handlefunc, as for change requested to move this function inside config.go currently the way i am doing this is loading the config file again and calling the scrape function for making the config changes available in next metric scrape.
```
log.Println("Parse config..")
if err := config.Load(configFile); err != nil {
	log.Fatal("Couldn't read ", *configFile, ": ", err)
}
if *decoupledScraping {
	s.decoupled(ctx)
} else {
	s.scrape(ctx)
}
```
What are your thought about this approach
